Remove references to v2 commands, remove unused functions, and constants

valaparthvi opened the pull request:
<!-- 
Thank you for opening a PR! Here are some things you need to know before submitting:

1. Please read our developer guideline: https://github.com/redhat-developer/odo/wiki/Dev:-odo-Dev-Guidelines
2. Label this PR accordingly with the '/kind' line
3. Ensure you have written and ran the appropriate tests: https://github.com/redhat-developer/odo/wiki/Dev:-Writing-and-running-tests
4. Read how we approve and LGTM each PR: https://github.com/redhat-developer/odo/wiki/Pull-Requests:-Review-guideline

Documentation:

If you are pushing a change to documentation, please read: https://github.com/redhat-developer/odo/wiki/Documentation:-Contributing
-->

**What type of PR is this:**
/kind code-refactoring
<!--
Add one of the following kinds:
/kind bug
/kind feature
/kind cleanup
/kind tests
/kind documentation

Feel free to use other [labels](https://github.com/redhat-developer/odo/labels) as needed. However one of the above labels must be present or the PR will not be reviewed. This instruction is for reviewers as well.
-->

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**
<!-- 
Specifying the issue will automatically close it when this PR is merged
-->

Fixes #5636 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
1. Used Goland editor inspection tool to identify unused functions, and constants.

feloy commented:
As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.

I would prefer to delete these ones when we are done with v3.

valaparthvi commented:
> As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.
> 
> I would prefer to delete these ones when we are done with v3.

Can you point out the functions? A few at least?

feloy commented:
> > As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.
> > I would prefer to delete these ones when we are done with v3.
> 
> Can you point out the functions? A few at least?

Here are some functions I think it would be interesting to keep for the moment:

- `GetProjectTypeFromDevfileMetadata`
- `GetLanguageFromDevfileMetadata` to be exhaustive on `Get*FromDevfileMetadata`
- `PreStart DevfileEventType = "preStart"` and other event types to be exhuastive
- all methods in `pkg/log/status.go` to have more choices on displaying output

valaparthvi commented:
> > > As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.
> > > I would prefer to delete these ones when we are done with v3.
> > 
> > 
> > Can you point out the functions? A few at least?
> 
> Here are some functions I think it would be interesting to keep for the moment:
> 
>     * `GetProjectTypeFromDevfileMetadata`
> 
>     * `GetLanguageFromDevfileMetadata` to be exhaustive on `Get*FromDevfileMetadata`
> 
>     * `PreStart DevfileEventType = "preStart"` and other event types to be exhuastive
> 
>     * all methods in `pkg/log/status.go` to have more choices on displaying output

I removed the Get*FromDevfileMetadata methods because they were added IIRC to add them to telemetry data, I do not believe they would be useful someplace else.

The devfile events have been around for a long time and they haven't been put to use, so I removed them. And they're anyway constants, we can add them back with v2 as a reference.

About methods in `pkg/log/status.go`, they've been around for quite sometime(even during v2)  in an unused state. The new commands have a different UI/UX, I figured we would write new methods as and when required.

feloy commented:
/lgtm

valaparthvi commented:
@feloy Can you lgtm again? I had to rebase.

dharmit commented on pkg/component/component.go:
```diff
@@ -35,26 +35,8 @@ func GetComponentTypeFromDevfileMetadata(metadata devfile.DevfileMetadata) strin
 }
 
 // GetProjectTypeFromDevfileMetadata returns component type from the devfile metadata
```
Please delete the comment for the function that was removed.

dharmit commented on pkg/component/component.go:
```diff
@@ -35,26 +35,8 @@ func GetComponentTypeFromDevfileMetadata(metadata devfile.DevfileMetadata) strin
-	return projectType
-}
 
 // GetLanguageFromDevfileMetadata returns component type from the devfile metadata
```
Please delete the comment for the function that was removed.

dharmit commented on pkg/devfile/image/image.go:
```diff
@@ -56,7 +57,7 @@ func BuildPushImages(devfileObj parser.DevfileObj, path string, push bool) error
 
 // BuildPushSpecificImage build an image defined in the devfile
 // If push is true, also push the image to its registry
-func BuildPushSpecificImage(devfileObj parser.DevfileObj, devfilePath string, component devfile.Component, push bool) error {
+func BuildPushSpecificImage(devfilePath string, component devfile.Component, push bool) error {
```
Can you update the function's comment since we are removing `devfileObj`?

dharmit reviewed (changes requested):
/approve

dharmit commented:
/lgtm

valaparthvi commented:
/retest-required

valaparthvi commented:
/override Kubernetes Integration Tests/Kubernetes Integration Tests

valaparthvi commented:
/override Unit Tests/Unit Tests
IBM test passes.
